src/hotspot/share/gc/serial/serialFullGC.cpp

```diff
@@ -280,6 +284,7 @@ class Compacter {
       _num_spaces = 3;
     }
     _index = 0;
+    _old_gen = heap->old_gen();
```

Comment:
This and the above initialization could be moved to the initializer list which is preferred.

Reply:
Since the array-part of init can't be moved to the initializer-list, I kept all init inside this body to avoid splitting inits up.